ClassMoodApp/static/js/controllers/LiveViewController.js

```diff
@@ -0,0 +1,118 @@
+                }
+                else {
+                    _this.getGauge(_this);
+                }
+            });
+        };
+        LiveViewController.prototype.getLectureId = function () {
+            this.lectureId = parseInt($("#lectureId").attr('lecture-id'), 10);
+        };
+        LiveViewController.prototype.votePace = function (newPace) {
+            this.gauge.pace_num = newPace;
+            this.postCurrentGauge();
+        };
+        LiveViewController.prototype.voteDepth = function (newDepth) {
+            this.gauge.depth_num = newDepth;
+            this.postCurrentGauge();
+        };
+        LiveViewController.prototype.getGauge = function (self) {
+            self.$http.get("/live_lecture/gauge/get/" + self.lectureId)
+                .success(function (response) {
+                self.gauge.pace_num = response.results[0];
+                if (self.gauge.pace_num == -1) {
+                    var pace = $("#pace1");
+                    $("#pace1").prop('checked', true);
+                    $("#pace1").parent().addClass('active');
+                }
+                else if (self.gauge.pace_num == 1) {
+                    $("#pace3").prop('checked', true);
+                    $("#pace3").parent().addClass('active');
+                }
+                else {
+                    $("#pace2").prop('checked', true);
+                    $("#pace2").parent().addClass('active');
+                }
+                self.gauge.depth_num = response.results[1];
+                if (self.gauge.depth_num == -1) {
+                    $("#depth1").prop('checked', true);
+                    $("#depth1").parent().addClass('active');
+                }
+                else if (self.gauge.depth_num == 1) {
+                    $("#depth3").prop('checked', true);
+                    $("#depth3").parent().addClass('active');
+                }
+                else {
+                    $("#depth2").prop('checked', true);
+                    $("#depth2").parent().addClass('active');
+                }
+            })
+                .error(function (response) {
+            });
+        };
+        LiveViewController.prototype.postCurrentGauge = function () {
+            this.$http.post('/live_lecture/gauge/update', JSON.stringify({ lect_id: this.lectureId, pace_num: this.gauge.pace_num, depth_num: this.gauge.depth_num }))
+                .success(function (data, status, headers, config) {
+            })
+                .error(function (data, status, header, config) {
+                console.log("ERROR while posting current gauge");
+            });
+        };
+        LiveViewController.prototype.gaugePollPace = function (self) {
+            this.$http.get("/live_lecture/pace/get/" + this.lectureId)
+                .success(function (response) {
+                self.gauge.pace_num = response;
+                $('#paceSlider').get(0).MaterialSlider.change(self.gauge.pace_num);
```

Comment:
Unfortunately I don't think this is possible. The material design sliders require the change() method be called to update the value instead of setting the value on the slider since it needs to run some of its own js to update the graphics. I can't seem to find a way to get ng-model or ng-bind to call a function with the updated value.